src/table/components/head/ColumnAdjuster.tsx

```diff
@@ -21,48 +23,84 @@ const ColumnAdjuster = ({ column, isLastColumn }: AdjusterProps) => {
 
   if (!applyColumnWidths || constraints.active) return null;
 
-  const mouseMoveHandler = (evt: MouseEvent) => {
-    const deltaWidth = evt.clientX - tempWidths.current.initX;
-    const adjustedWidth = Math.max(tempWidths.current.initWidth + deltaWidth, MIN_COLUMN_WIDTH);
-
+  const updateWidth = (adjustedWidth: number) => {
     tempWidths.current.columnWidth = adjustedWidth;
     const newColumnWidths = [...columnWidths];
     newColumnWidths[pageColIdx] = adjustedWidth;
     setColumnWidths(newColumnWidths);
   };
 
-  const mouseUpHandler = (evt: MouseEvent) => {
-    evt.preventDefault();
-    evt.stopPropagation();
-    document.removeEventListener('mousemove', mouseMoveHandler);
-    document.removeEventListener('mouseup', mouseUpHandler);
-
+  const confirmWidth = () => {
     if (tempWidths.current.columnWidth !== tempWidths.current.initWidth) {
       const newWidthData = { type: ColumnWidthTypes.PIXELS, pixels: tempWidths.current.columnWidth };
       applyColumnWidths(newWidthData, column);
     }
   };
 
+  const mouseMoveHandler = (evt: MouseEvent) => {
+    const deltaWidth = evt.clientX - tempWidths.current.initX;
+    const adjustedWidth = Math.max(tempWidths.current.initWidth + deltaWidth, MIN_COLUMN_WIDTH);
+    updateWidth(adjustedWidth);
+  };
+
+  const mouseUpHandler = (evt: MouseEvent, adjuster: HTMLDivElement) => {
+    preventDefaultBehavior(evt);
+    document.removeEventListener('mousemove', mouseMoveHandler);
+    document.removeEventListener('mouseup', () => mouseUpHandler(evt, adjuster));
+
+    confirmWidth();
+    adjuster.blur();
+  };
+
   const mouseDownHandler = (evt: React.MouseEvent) => {
     // evt.preventDefault();
     evt.stopPropagation();
+    const adjuster = evt.target as HTMLDivElement;
 
     tempWidths.current = {
       initX: evt.clientX,
       initWidth: columnWidths[pageColIdx],
       columnWidth: columnWidths[pageColIdx],
     };
+
     document.addEventListener('mousemove', mouseMoveHandler);
-    document.addEventListener('mouseup', mouseUpHandler);
+    document.addEventListener('mouseup', (event) => mouseUpHandler(event, adjuster));
```

Comment:
I don't think this will remove the listener properly later. Pretty sure it has to be the same instance of the callback for the remover to find it

Reply:
Fixed.